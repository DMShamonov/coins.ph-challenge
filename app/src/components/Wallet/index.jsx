// @flow

import * as React from 'react';
import { Link } from 'react-router-dom';

import ROUTES from 'constants/routes';
import { TableCell, TableRow } from 'components/Table';
import Value from 'components/Value';
import { calcBtc } from 'controllers/utilities';

import type { PropsType } from './types.js.flow';

export default function Wallet({
  id,
  label,
  balance,
}: PropsType): React.Node {
  return (
    <TableRow>
      <TableCell><Link to={ROUTES.WALLET_TRANSACTIONS.replace(':id', id)}>{id}</Link></TableCell>
      <TableCell>{label}</TableCell>
      <TableCell width="10%">
        <Value value={calcBtc(balance)} />
      </TableCell>
    </TableRow>
  );
}
