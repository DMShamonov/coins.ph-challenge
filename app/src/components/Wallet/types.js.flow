// @flow

export type PropsType = {|
  id: string,
  label: string,
  balance: number,
|};
