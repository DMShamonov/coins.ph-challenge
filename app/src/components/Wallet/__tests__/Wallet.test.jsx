import React from 'react';
import { shallow } from 'enzyme';

import Wallet from '../index';

describe('`Wallet` component should render correctly', () => {
  test('with default props', () => {
    expect(shallow(<Wallet id="__ID__" label="__LABEL__" balance={100000000} />)).toMatchSnapshot();
  });
});
