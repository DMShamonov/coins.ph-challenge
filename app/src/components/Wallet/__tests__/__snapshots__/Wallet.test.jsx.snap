// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`\`Wallet\` component should render correctly with default props 1`] = `
<TableRow>
  <TableCell>
    <Link
      to="/wallets/__ID__/transactions"
    >
      __ID__
    </Link>
  </TableCell>
  <TableCell>
    __LABEL__
  </TableCell>
  <TableCell
    width="10%"
  >
    <Value
      value={1}
    />
  </TableCell>
</TableRow>
`;
